### PR TITLE
[Python] Concepts naming cleanup and typo patrol

### PR DESCRIPTION
**Pull request by BethanyG**
*  Edited `reference/README.md` & `exercises/README.md` to change the class concept names and add numbers exercise.
*  Edited `list` and `list-method` `about.md` docs for typos, code examples, and style
*  Edited `numbers` `about.md` for better `float` type description/example.
* Edited Elyses Enchantments (lists) and Chaitanas Colossal Coaster (list-methods) to make them closer to the `about.md` docs in examples and style.

### CODE REVIEW

**Review by valentin-p — APPROVED** [review]

File: languages/python/concepts/lists/about.md [hunk 1]
```diff
@@ -120,10 +120,9 @@ Items inside lists (_like the sequence types `string` and `tuple`_), can be acce
 'Oatmeal'
 ```
 
-You can access a portion of a list with _slice notation_ (`[start:stop]`). A _slice_ is defined as the sequence of items in a list at position `index` such that `start <= index < stop`.
+You can also access a portion of a list with _slice notation_ (`[start:stop]`). A _slice_ is defined as the sequence of items in a list at position `index` such that `start <= index < stop`. _Slicing_ does not modify the original `list`. Instead, you get a new list with copies of all the elements you asked for.
```
Comment (valentin-p) [inline languages/python/concepts/lists/about.md:123]:
```suggestion
You can also access a portion of a list with _slice notation_ (`[start:stop]`). A _slice_ is defined as the sequence of items in a list at position `index` such that `start <= index < stop`. _Slicing_ does not modify the original `list`. Instead, you get a new list with (only) the elements you asked for.
```

File: languages/python/concepts/numbers/about.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ Each of these `int` displays can be converted into the other via constructor:
 1234
 ```
 
-Numbers containing a decimal point (_with or without any numbers following_) are identified as `floats`:
+Numbers containing a decimal point (with or without fractional parts) are identified as `floats`:
```
Comment (valentin-p) [inline languages/python/concepts/numbers/about.md:68]: Thanks!

File: languages/python/exercises/concept/elyses-enchantments/.docs/introduction.md [hunk 3]
```diff
@@ -104,18 +110,20 @@ Items inside lists (_like the sequence types `string` and `tuple`_), can be acce
...
 ```python
 >>> colors = ["Red", "Purple", "Green", "Yellow", "Orange", "Pink", "Blue", "Grey"]
+
 # If there is no step parameter, the step is assumed to be 1.
 >>> middle_colors = colors[2:6]
 ["Green", "Yellow", "Orange", "Pink"]
-# If the stop parameter is omitted, the slice will stop at the end of the list.
->>> primary_colors = colors[0::3]
+
+#If the start or stop parameters are omitted, the slice will
```
Comment (valentin-p) [inline languages/python/exercises/concept/elyses-enchantments/.docs/introduction.md:124]:
```suggestion
# If the start or stop parameters are omitted, the slice will
```

File: languages/python/concepts/numbers/about.md (hunk 2 above)
Comment (BethanyG) [reply languages/python/concepts/numbers/about.md:68]: 😄

**Pull request merged**